src/v/cluster/cloud_metadata/cluster_recovery_backend.cc

```diff
@@ -366,7 +367,7 @@ cluster_recovery_backend::find_controller_snapshot_in_bucket(
             const size_t snap_size = co_await reader.get_snapshot_size();
             auto snap_buf_parser = iobuf_parser{
               co_await read_iobuf_exactly(reader.input(), snap_size)};
-            snapshot = serde::read<cluster::controller_snapshot>(
+            snapshot = co_await serde::read_async<cluster::controller_snapshot>(
```

Comment:
Side note, but it would be great to somehow explicitly mark sync serde forbidden for the snapshot struct and not rely on is_standard_layout implicitly being false.